src/services/api/me/me.service.ts

```diff
@@ -48,4 +58,82 @@ export class MeService {
 
     return this.spaceService.getSpaces(args);
   }
+
+  public async getMyJourneys(
+    agentInfo: AgentInfo,
+    limit = 20,
+    visibilities: SpaceVisibility[] = [
+      SpaceVisibility.ACTIVE,
+      SpaceVisibility.DEMO,
+    ]
+  ): Promise<MyJourneyResults[]> {
+    const credentialMap = groupCredentialsByEntity(agentInfo.credentials);
+    const spaceIds = Array.from(credentialMap.get('spaces')?.keys() ?? []);
+    const challengeIds = Array.from(
+      credentialMap.get('challenges')?.keys() ?? []
+    );
+    const opportunityIds = Array.from(
+      credentialMap.get('opportunities')?.keys() ?? []
+    );
+    const args: SpacesQueryArgs = {
+      IDs: spaceIds,
+      filter: {
+        visibilities,
+      },
+    };
+
+    const spaces = await this.spaceService.getSpaces(args, {
+      relations: { collaboration: true },
+    });
+    const challenges = await this.challengeService.getChallenges({
+      where: { id: In(challengeIds) },
+      relations: {
+        collaboration: true,
+        innovationFlow: true,
+        parentSpace: true,
+      },
+    });
+    const opportunities = await this.opportunityService.getOpportunities({
+      where: { id: In(opportunityIds) },
+      relations: { collaboration: true, innovationFlow: true, challenge: true },
+    });
+
+    const myJourneyResults: MyJourneyResults[] = [];
+
+    // Process spaces, challenges, and opportunities together
+    const entitiesToProcess = [...spaces, ...challenges, ...opportunities];
+
+    for (const entity of entitiesToProcess) {
+      if (!entity.collaboration?.id) continue;
```

Comment:
this sound like a serious issue. Might be good to raise `EntityNotInitializedException` or log it.